### PR TITLE
New issues

### PR DESCRIPTION
**Pull request by ssadat**

### CODE REVIEW

**Comment by kibaekkim** [comment]
Need to add test and update example

**Comment by ssadat** [comment]
How to add a test to check if all outputs have been suppressed? Do we need a test for it?

File: ext_solver.test [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+print(model);
+println("________________________________________");
+
+if (typeof(optimizer) == MOI.OptimizerWithAttributes) status = try JuMP.optimize!(model); termination_status(model) catch; nothing end end
+
+if status != nothing 
+xsol = JuMP.value.(X)
+ysol = JuMP.value.(Y)
+end
+
```
Comment (kibaekkim) [inline ext_solver.test:38]: Remove this file as `test/ext_solver.test` already exists.

File: src/parameters.jl [hunk 2]
```diff
@@ -4,10 +4,13 @@ Base.@kwdef mutable struct Parameters
     algorithm::String = "Line Search" # Defines the algorithm -- either Line Search or Trust Region Method (Trust Region algorithm hasn't been implemented yet)
 
     # Defines the external solver for suproblems
-    external_optimizer::Union{Nothing,DataType} = nothing
+    external_optimizer::Union{Nothing,DataType,MOI.OptimizerWithAttributes} = nothing
```
Comment (kibaekkim) [inline src/parameters.jl:7]: Do we allow `DataType` type too? What if `external_optimizer` is given without attribute?

File: src/algorithms/slp_line_search.jl [hunk 3]
```diff
@@ -49,7 +49,7 @@ mutable struct SlpLS{T,Tv,Tt} <: AbstractSlpOptimizer
         slp.alpha = 1.0
 
         slp.options = problem.parameters
-        slp.optimizer = slp.options.external_optimizer()
+        slp.optimizer = MOI.instantiate(slp.options.external_optimizer)
```
Comment (kibaekkim) [inline src/algorithms/slp_line_search.jl:52]: Does this still work if `external_optimizer::DataType`?

**Review by kibaekkim — CHANGES_REQUESTED** [review]
Please address the comments.

File: src/parameters.jl [hunk 4]
```diff
@@ -4,10 +4,13 @@ Base.@kwdef mutable struct Parameters
...
     
     # Whether to use approximation hessian (limited-memory), exact, or none
     hessian_type::String = "none"
+    
+    # Output flag to show the output message
+    OutputFlag::Int = 1	   # 0 supresses all outputs excep warnings and errors
```
[the remark this answers is not in the document]
Comment (ssadat) [reply src/parameters.jl:13]: #98 created

File: test/runtests.jl [hunk 5]
```diff
@@ -1,15 +1,28 @@
+push!(LOAD_PATH, "/home/ssadat/projects/Nonlinear_Optimization/ActiveSetMethods/src")
```
[the remark this answers is not in the document]
Comment (ssadat) [reply test/runtests.jl:1]: #99 created

File: src/parameters.jl (hunk 2 above)
Comment (ssadat) [reply src/parameters.jl:7]: That's why we should allow `DataType`. If we remove `DataType`, `external_optimizer` without `attributes` will give error.

File: src/algorithms/slp_line_search.jl (hunk 3 above)
Comment (ssadat) [reply src/algorithms/slp_line_search.jl:52]: Yes. `MOI.instantiate` works for both `DataType` and `MOI.OptimizerWithAttributes`

File: test/runtests.jl [hunk 6]
```diff
@@ -1,15 +1,28 @@
...
 #     include("unittests.jl")
 # end
 
+
+@testset "MathOptInterface" begin
+    include("MOI_wrapper.jl")
+end
+
+@testset "External Solver Attributes Implementation with Toy Example" begin
+    include("ext_solver.test")
```
[the remark this answers is not in the document]
Comment (ssadat) [reply test/runtests.jl:19]: #100 created

File: test/ext_solver.test [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+optimizer_solver = nothing
+model = nothing
+xsol = 0.0
+ysol = 0.0
+status = nothing
+
+optimizer_solver = try optimizer_with_attributes(ActiveSetMethods.Optimizer,"external_optimizer"=>optimizer_with_attributes(GLPK.Optimizer,"msg_lev"=>GLPK.MSG_OFF)); catch; ActiveSetMethods.Optimizer; end
+
+model = try Model(optimizer_solver); catch; Model(ActiveSetMethods.Optimizer); end
```
[the remark this answers is not in the document]
Comment (ssadat) [reply test/ext_solver.test:14]: #101 created

File: ext_solver.test (hunk 1 above)
Comment (ssadat) [reply ext_solver.test:38]: #102 created

**Review by kibaekkim — APPROVED** [review]

**Pull request merged**